runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -542,13 +543,16 @@ setCallSiteTargetImpl(J9VMThread *currentThread, jobject callsite, jobject targe
 		MM_ObjectAccessBarrierAPI objectAccessBarrier = MM_ObjectAccessBarrierAPI(currentThread);
 
 		/* Check for MutableCallSite modification. */
-		J9JITConfig* jitConfig = currentThread->javaVM->jitConfig;
-		J9UTF8 *clazzNameUTF8 = J9ROMCLASS_CLASSNAME(clazz->romClass);
-		U_8 *clazzName = J9UTF8_DATA(clazzNameUTF8);
-		UDATA clazzLength = J9UTF8_LENGTH(clazzNameUTF8);
+		J9JITConfig* jitConfig = javaVM->jitConfig;
+		J9Class *mcsClass = vmFuncs->hashClassTableAt(
+			javaVM->systemClassLoader,
+			(U_8*)"java/lang/invoke/MutableCallSite",
+			LITERAL_STRLEN("java/lang/invoke/MutableCallSite"));
```

Comment:
This seems like a good candidate for the vmconstantpool.xml to cache the class for us under the `opt_methodHandleCommon` flag.  It's already pre-loaded for j9-impl of MHs

Reply:
I had suggested this approach since we only resolve knownclasses in the JCL constantPool if they have known fields/methods (`initializeKnownClasses`). `J9VMJAVALANGINVOKEMUTABLECALLSITE_OR_NULL` kept returning NULL as there are no known fields/methods for MCS in the ojdk_mh version.

The alternative is to use `J9VMJAVALANGINVOKEMUTABLECALLSITE` (`resolveKnownClass`) but that will trigger a classload which I was hoping to avoid here.

Reply:
I guess the 3rd option is to add a known field to MCS in the ojdk_mh version.

Reply:
@DanHeidinga I had a discussion with Devin. The 3rd option seems best, its a bit of a hack since we wont actually be using the known field.

Reply:
Ideally here I would want to get the class from the VM constant pool if it's been loaded, and otherwise null is fine, since if the class hasn't been loaded, then we don't have an instance of it. But that doesn't seem to be an option for the VM constant pool - apparently we have to either add a known field to make the VM load the class eagerly, or else trigger a load here with whatever complications that involves

Reply:
Thanks for clarifying the vmconstantpool behaviour.  

The 3rd option is really hacky though.  Let's not go down that path as there's functional code here.  Better to keep what we have here then to add a dummy field to force the load.  (Though if we OK with forcing the load, then I'm not sure why `J9VMJAVALANGINVOKEMUTABLECALLSITE` forcing the load is an issue)

Reply:
>  Better to keep what we have here then to add a dummy field to force the load.

Just to be clear, MCS has a field, so we wont be adding a new field to the class. We would just be adding an entry for that field in the vmconstantpool.xml, but not using the generated macros for it.

> Though if we OK with forcing the load, then I'm not sure why J9VMJAVALANGINVOKEMUTABLECALLSITE forcing the load is an issue

I would prefer to force the load at startup, rather than here in `setCallsiteTargetImpl` as it means additional complexity in keeping the `target` and `callsite` objects visible to the GC.

Reply:
To be clear, I'm OK with @jdmpapin existing code given the clarification of the vmconstantpool's behaviour.